text/language-bindings/README.md

```diff
@@ -0,0 +1,196 @@
+using System.Text;
+using System.Runtime.InteropServices;
+
+using Hyperledger.Ursa.Api;
+
+namespace Bls
+{
+    public static class Bls
+    {
+        private const int SIGNATURE_BYTES = 32;
+
+        public function byte[] Sign(string message, byte[] privateKey)
+        {
+            return Sign(Encoding.UTF8.GetBytes(message), private_key);
+        }
+
+        public function byte[] Sign(byte[] message, byte[] privateKey)
+        {
+            var signature = new byte[SIGNATURE_BYTES];
+
+            if (NativeMethods.ursa_bls_sign(out signature, SIGNATURE_BYTES, message, message.Length, privateKey, privateKey.Length) == 0)
+            {
+                return signature;
+            }
+            throw new Exception("An error occurred while signing message");
+        }
+
+        public function bool Verify(string message, byte[] publicKey, byte[] signature)
+        {
+            return Verify(Encoding.UTF8.GetBytes(message), publicKey, signature);
+        }
+
+        public function bool Verify(byte[] message, byte[] publicKey, byte[] signature)
+        {
+            switch (NativeMethods.ursa_bls_verify(message, message.Length,
+                                                  privateKey, privateKey.Length,
+                                                  signature, signature.Length))
+            {
+                case 0: return true;
+                default: return false;
+            }
+        }
+    }
+}
+```
+
+The idiomatic wrapper provides more logic and parameters for convenience than the thin wrapper does.
+It can also be expanded to include other types but ultimately maps all inputs to
+the expected thin wrapper types.
+
+Thin wrappers are composed of many functions using the [foreign function interface (FFI)](https://doc.rust-lang.org/nomicon/ffi.html).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Thin wrappers can be problematic to write by hand and maintain as APIs change.
+It is desirable to have thin wrappers generated programmatically.
+Rust supports generating FFI using `bindgen`. `bindgen` can map to other languages like
+C and WASM. Thus thin wrappers should be created and maintained as much as possible using features like `bindgen`.
+Thin wrappers should be included in the Ursa project itself in a folder called `wrappers`. This will allow them to
+stay in-sync with any other changes to the core library.
+
+Idiomatic wrappers usually cannot be generated by hand as they require more in depth language experience
+than automatic generators can produce. Therefore, idiomatic wrappers should be written by project contributors
+and maintained in separate repositories. These wrappers also SHOULD include tests for interacting with
+native Ursa.
+
+Here are general guidelines for producing wrappers:
+
+`<language>` should follow the naming convention as described [here](https://support.codebasehq.com/articles/tips-tricks/syntax-highlighting-in-markdown)
```

Comment:
Is the link appropriate? Where are the naming conventions?

Reply:
Looks like this is marked as resolved but the link isn't changed?

Reply:
At the link, you should see after the second code section `Here's a full list of supported languages:`. Following that is a list of abbreviated languages. That is what I am referring to.

Reply:
Ok, so this is the naming convention for paths/repos for languages, i.e. not java-script but js.